include/nbl/builtin/hlsl/prefix_sum_blur/box_sampler.hlsl

```diff
@@ -127,36 +138,36 @@ struct BoxSampler
             }
             case ETC_CLAMP_TO_EDGE:
             {
-                result -= (1 - abs(leftIdx)) * prefixSumAccessor.template get<T, uint32_t>(0);
+                result -= (leftIdx + 1) * prefixSumAccessor.template get<T, uint32_t>(0);
                 break;
             }
             case ETC_MIRROR:
             {
                 const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
                 T floored, ceiled;
 
-                if (abs(leftFlIdx + 1) % (2 * linearSize) == 0)
-                    floored = -(abs(leftFlIdx + 1) / linearSize) * last;
+                if (abs(leftFlIdx) % (2 * linearSize) == 0)
+                    floored = -(abs(leftFlIdx) / linearSize) * last;
                 else
                 {
-                    const uint32_t period = uint32_t(ceil(float32_t(abs(leftFlIdx + 1)) / linearSize));
+                    const uint32_t period = uint32_t(ceil(float32_t(abs(leftFlIdx)) / linearSize));
                     if ((period & 0x1u) == 1)
-                        floored = -(period - 1) * last - prefixSumAccessor.template get<T, uint32_t>((abs(leftFlIdx + 1) - 1) % linearSize);
+                        floored = -(period - 1) * last - prefixSumAccessor.template get<T, uint32_t>((abs(leftFlIdx) - 1) % linearSize);
                     else
-                        floored = -(period - 1) * last - (last - prefixSumAccessor.template get<T, uint32_t>((leftFlIdx + 1) % linearSize - 1));
+                        floored = -(period - 1) * last - (last - prefixSumAccessor.template get<T, uint32_t>(leftFlIdx % linearSize - 1));
                 }
 
                 if (leftClIdx == 0) // Special case, wouldn't be possible for `floored` above
                     ceiled = 0;
-                else if (abs(leftClIdx + 1) % (2 * linearSize) == 0)
-                    ceiled = -(abs(leftClIdx + 1) / linearSize) * last;
+                else if (abs(leftClIdx) % (2 * linearSize) == 0)
+                    ceiled = -(abs(leftClIdx) / linearSize) * last;
                 else
                 {
-                    const uint32_t period = uint32_t(ceil(float32_t(abs(leftClIdx + 1)) / linearSize));
+                    const uint32_t period = uint32_t(ceil(float32_t(abs(leftClIdx)) / linearSize));
                     if ((period & 0x1u) == 1)
-                        ceiled = -(period - 1) * last - prefixSumAccessor.template get<T, uint32_t>((abs(leftClIdx + 1) - 1) % linearSize);
+                        ceiled = -(period - 1) * last - prefixSumAccessor.template get<T, uint32_t>((abs(leftClIdx) - 1) % linearSize);
                     else
-                        ceiled = -(period - 1) * last - (last - prefixSumAccessor.template get<T, uint32_t>((leftClIdx + 1) % linearSize - 1));
+                        ceiled = -(period - 1) * last - (last - prefixSumAccessor.template get<T, uint32_t>(leftClIdx % linearSize - 1));

```

Comment:
all your `abs(leftClIdx)` and `abs(leftFlIdx)` could be `-leftClIfx` and `leftFlIdx` respectively because `leftIdx<0` and the highest you can ceil to is 0